### PR TITLE
Document gcsweb

### PR DESCRIPTION
**Pull request by techtonik**
Please review. It would be also nice to get link to deployed gcsweb to see how it looks like.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/k8s.io/22)

### CODE REVIEW

**Comment by thockin** [comment]
http://gcsweb.k8s.io/gcs/kubernetes-release/release/

Let me know when you're done editing and I will review.

**Comment by techtonik** [comment]
@thockin done with comments from https://github.com/kubernetes/k8s.io/issues/23#issuecomment-252809908

File: gcsweb/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+`kubernetes` releases can be downloaded using direct API links to specific
+files. However, to browse all available files at
+https://console.cloud.google.com/storage/browser/kubernetes-release/release/
+or with `gsutil` people
+[need Google login](https://cloud.google.com/storage/docs/access-public-data).
+
+#### Solution
+
+Run a public web app that uses authenticated requests on backend to extract
```
Comment (thockin) [inline gcsweb/README.md]: This does not use authenticated requests - it uses the public API to access public buckets.

**Comment by thockin** [comment]
just one nit, thanks.

File: gcsweb/README.md (hunk 1 above)
Comment (techtonik) [inline gcsweb/README.md]: Done.

**Comment by techtonik** [comment]
I still wonder where is the source code?

**Comment by thockin** [comment]
The source is in github.com/kubernetes/test-infra/gcsweb :)

Also, I just caused you to need a rebase, though github doesn't show here - I renamed gcsweb to gcsweb.k8s.io.  Sorry.

File: gcsweb/README.md [hunk 2]
```diff
@@ -24,3 +24,11 @@ links to specific files.
 2. `gcsweb` is not designed for initial browsing (yet?) - it doesn't list
 which GCS buckets are available, and because bucket is a part of URL, you
 need a documented link to browse.
+
+#### Installation and deployment
+
+`gcsweb` is built from source code at
+http://github.com/kubernetes/test-infra/gcsweb into Docker container, which
```
Comment (thockin) [inline gcsweb/README.md]: That is not a valid URL - it was a go-path in email.  sorry.  https://github.com/kubernetes/test-infra/tree/master/gcsweb

File: gcsweb/README.md [hunk 3]
```diff
@@ -24,3 +24,11 @@ links to specific files.
...
+is then uploaded to private container storage at
+https://gcr.io/google_containers/gcsweb-amd64 and fetched during processing
```
Comment (thockin) [inline gcsweb/README.md]: I appreciate the docs, but this last part is pretty self-evident from reading the deployment yaml :)

**Comment by techtonik** [comment]
Added link to sources. What is the command to deploy the application? Might be obvious, and I suspect that it is `kubectl`, but I don't have any cluster to experiment with just yet.

Rebasing needs some time as I edit this from the web, so if I won't respond in a couple of hours, feel free to merge as-is.

**Comment by thockin** [comment]
All of these are managed with `kubectl apply`.  If this is done from the web ui, you might just close the PR and open another in the new directory.  Easier.

File: gcsweb/README.md (hunk 2 above)
Comment (techtonik) [inline gcsweb/README.md]: Yes, fixed that.

File: gcsweb/README.md (hunk 3 above)
Comment (techtonik) [inline gcsweb/README.md]: Is this repository aims at providing best practices https://github.com/kubernetes/k8s.io/issues/23#issue-182170987 then documenting things like `kubectl apply` is necessary. Reading small paragraph here may save hours of various tutorials elsewhere. I would record video with example session from scratch (given that cluster is already there), but as I said I don't have any cluster to experiments (no credit card), so it is a pure theory for me.

**Comment by techtonik** [comment]
Done rebasing.

**Comment by thockin** [comment]
LGTM

**Comment by thockin** [comment]
The CLA bot is not running on this repo - have you signed the CLA?  Under what name?

**Comment by techtonik** [comment]
@thockin signed, yes. Check bots here - https://github.com/kubernetes/md-check/pull/7

**Pull request merged**